check_patroni/cluster.py

```diff
@@ -83,35 +83,96 @@ def __init__(self, connection_info: ConnectionInfo, max_lag: Union[int, None]):
         self.max_lag = max_lag
 
     def probe(self) -> Iterable[nagiosplugin.Metric]:
-        item_dict = self.rest_api("cluster")
+        def find_leader_timeline(json: Any) -> int:
+            for member in json:
+                if member["role"] == "leader":
+                    leader_tl = str(member["timeline"])
+                    break
```

Comment:
There should be an `else:` clause in case the `if` never breaks the loop, otherwise, `leader_tl` will be unbound and this will fail with a `NameError` below. The `else:` could raise a `ValueError`, to be caught by caller or return `None` depending on the semantics.

Reply:
I think it should return none,  it's possible to have no leader in some cases.
In that case I suppose it's fine to say there is no "healthy replica" since the 
cluster is not working properly.

I'll fix this and add a test case for the same.